x-pack/plugins/cases/public/components/all_cases/table.tsx

```diff
@@ -88,7 +88,7 @@ export const CasesTable: FunctionComponent<CasesTableProps> = ({
       <EuiLoadingContent data-test-subj="initialLoadingPanelAllCases" lines={10} />
     </Div>
   ) : (
-    <Div>
+    <Div data-test-subj={isCasesLoading ? 'cases-table-loading' : null}>
```

Comment:
I don't like the idea of dynamic `data-test-subj`. This creates an ambiguity of the state of the page. Why don't use the `data-test-subj` of the spinner?

Reply:
the spinner doesn't have a data-test-subj nor we can set one. @pheyos gave me this idea and it is how they do it in other places.

Reply:
Yes, we do something similar on the ML / Transform side, see e.g. [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/public/application/jobs/jobs_list/components/jobs_list/jobs_list.js#L405). Sometimes, there's no loading indicator / no good alternative to let the test know that some loading is done. And since the `data-` attributes `need not to have any defined meaning` ([source](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)), I think it's fair to also sometimes let them reflect some state. Personally, I'd prefer to avoid the `null` here and always assign some value, that's changing based on loading state, but I guess that's a matter of taste.

Reply:
Ok! I am mostly concerned if this will introduce flakiness to the tests. If it is tested in other tests then I think we should be good.

Reply:
This was actually introduced to reduce flakiness and we fixed that flakiness thanks to this.

Reply:
Amazing!